nexus/tests/integration_tests/disks.rs

```diff
@@ -2594,6 +2595,249 @@ async fn test_disk_expunge(cptestctx: &ControlPlaneTestContext) {
     assert_eq!(expunged_regions.len(), 3);
 }
 
+#[nexus_test(extra_sled_agents = 3)]
+async fn test_do_not_provision_on_dataset(cptestctx: &ControlPlaneTestContext) {
+    let nexus = &cptestctx.server.server_context().nexus;
+    let datastore = nexus.datastore();
+    let opctx =
+        OpContext::for_tests(cptestctx.logctx.log.new(o!()), datastore.clone());
+
+    // Create one zpool, each with one dataset, on all the sleds
+    let disk_test = DiskTestBuilder::new(&cptestctx)
+        .on_all_sleds()
+        .with_zpool_count(1)
+        .build()
+        .await;
+
+    // For one of the datasets, mark it as not provisionable
```

Comment:
How many datasets total does this test create.

Reply:
Four - one per pool, where there's one pool per agent.